### DISCUSSION TITLE
How to run `wingetcreate new` non-interactively?

### DISCUSSION THREAD

**Discussion by ldennington** [microsoft/winget-create#190] [Q&A] [answered]
### Discussed in https://github.com/microsoft/winget-create/discussions/188

Creating an issue for visibility.

<div type='discussions-op-text'>

<sup>Originally posted by **ldennington** October 14, 2021</sup>
Hello,

I'm reaching out because I'm interested in using `wingetcreate` in my GitHub Actions workflows to publish packages my team owns to `winget`. We are currently using a custom task (see example [here](https://github.com/microsoft/Git-Credential-Manager-Core/blob/6c0fb9efed498ada7d1a76aff4daf8ec43f5c49a/.github/workflows/release-winget.yaml#L10-L38)), which has been a pain point since the manifest specification changes frequently. 

In light of this, I'm reaching out to see whether it's possible to run `wingetcreate new` non-interactively. Whenever I run any of the following locally, I'm prompted to manually accept/override the default values from my installer:

```
wingetcreate new https://github.com/ldennington/Git-Credential-Manager-Core/releases/download/winget_test/gcmcore-win-x86-2.0.559.38027.exe --out manifests --token <insert-token-here>
```

```
wingetcreate new https://github.com/ldennington/Git-Credential-Manager-Core/releases/download/winget_test/gcmcore-win-x86-2.0.559.38027.exe
```

I see [the example](https://github.com/microsoft/winget-create#using-windows-package-manager-manifest-creator-in-a-cicd-pipeline) shows the `wingetcreate update` command in a pipeline, but not `wingetcreate new`, which prompted me to ask this question.</div>

**Comment by jedieaston** [comment, marked as answer]
I don't think it's possible ATM. There's too much metadata that needs to be filled out (privacy policy, website, localized names, etc). `wingetcreate new` is designed to be ran only once for an application, whereas `wingetcreate update` is designed to be ran for every new version of an app (where it copies most of the values asked for during `new` from the previous version), as such, it's acceptable that it's not automated (or at least, it has been up to this point).

What's the usecase where you want to automate `wingetcreate new`? For the example action you linked, right now that could be replicated by the command `wingetcreate update -u {{url}} -v {{version}} Microsoft.GitCredentialManagerCore` (and if you want to submit, add the `--submit --token <TOKEN>` arguments to that).

**Reply by ldennington** [reply]
Thank you! I was confused on that since a new manifest is created for each new release, but this solution seems to have unblocked me on this particular issue. I'm still unable to use `wingetcreate` for GCM Core due to #166 but it's probably best to confine further discussions to that issue.
